python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -438,6 +436,86 @@ def create_id_map(ids):
     """
     return {id1: i for i, id1 in enumerate(ids)}
 
+def worker_fn(task_queue, res_queue, user_parser):
+    """ The worker function in the worker pool
+
+    Parameters
+    ----------
+    task_queue : Queue
+        The queue that contains all tasks
+    res_queue : Queue
+        The queue that contains the processed data. This is used for
+        communication between the worker processes and the master process.
+    user_parser : callable
+        The user-defined function to read and process the data files.
+    """
+    try:
+        while True:
+            # If the queue is empty, it will raise the Empty exception.
+            i, in_file = task_queue.get_nowait()
+            data = user_parser(in_file)
+            res_queue.put((i, data))
+            gc.collect()
```

Comment:
Why we need a gc collect?

Reply:
i try to trigger garbage collection as much as possible.